src/augmented_block.jl

```diff
@@ -57,7 +58,7 @@ function AugmentedNLPBlockModel(
         (block_hess_struct[1], block_hess_struct[2]),
         ATA,
     )
-    return AugmentedNLPBlockModel(meta, Counters(), nlp, λ, ρ, A, b, sol, hess_struct)
+    return AugmentedNLPBlockModel(meta, Counters(), nlp, λ, ρ, A, b, sol, sol, hess_struct)
```

Comment:
why do we need two `sol` here?

Reply:
The second one is to initialize `local_sol` with the current solution, which gets updated as every block is solved in an iteration of some decomposition algorithm.
 
The first `sol` gets updated only once all the blocks have been solved.

Reply:
Yeah, you were right. We do not need two `sol` for `AugmentedNLPBlockModel`. I have updated the code.

We do need an additional vector for the `ProxAugmentedNLPBlockModel` to be able to compute the proximal term.